include/psa/crypto_extra.h

```diff
@@ -1901,6 +1901,342 @@ static inline struct psa_pake_operation_s psa_pake_operation_init( void )
+ *                              computation, #PSA_ERROR_IN_PROGRESS will be
+ *                              returned by the function performing the
+ *                              computation. It is then the caller's
+ *                              responsibility to either call again with the
+ *                              same parameters until it returns 0 or an error
+ *                              code; or to free the relevant context if the
+ *                              operation is to be aborted.
+ *
+ * \note                        It is strictly required that all input
+ *                              parameters and the restart context be the same
+ *                              on successive calls for the same operation,
+ *                              but output parameters need not be the same;
+ *                              they must not be used until the function
+ *                              finally returns 0.
+ *
+ *                              This only applies to functions whose
+ *                              documentation mentions they may return
+ *                              #PSA_ERROR_IN_PROGRESS.
+ *
+ * \param max_ops               The maximum number of basic operations to
+ *                              be done in a row. Passing zero here has the
+ *                              same effect as passing the maximum value - i.e
+ *                              the operation will be done in one go. Lower
+ *                              (non-zero) values mean signature functions
+ *                              will block for a lesser maximum amount of time
+ *                              and conversely larger values will mean
+ *                              blocking for a larger maximum amount of time.
+ *                              The effect of changing this value will depend
+ *                              on the key used, the curve used and the
+ *                              underlying implementation, so we provide /c
+ *                              psa_sign_interruptible_get_num_ops() in order
+ *                              to help with tuning this value.
+ *
+ */
+void psa_sign_interruptible_set_max_ops( size_t max_ops );
+
+/**
+ * \brief                       Get the number of basic operations that an
+ *                              operation took to complete.
+ *
+ *                              This is a helper provided to help you tune the
+ *                              value passed to \c
+ *                              psa_sign_interruptible_set_max_ops().
+ *
+ *                              This only applies to functions whose
+ *                              documentation mentions they may return
+ *                              #PSA_ERROR_IN_PROGRESS.
+ *
+ * \param ctx                   The \c psa_sign_interruptible_context to use.
+ *                              This must be initialized first.
+ *
+ */
+size_t psa_sign_interruptible_get_num_ops(
```

Comment:
When am I allowed to call this? When the operation's in progress, or only once it has completed?

Reply:
Note: a "basic operation" (currently, modular multiplication) is not going to take the same amount of time for different curves. So, if the idea is that people can measure the time it takes an operation to complete, and divide by the number of basic operations as indicated by this function in order to get the time a "basic operation" represents, they should do this for each curve they might use.

Reply:
Forgot to mention: I like the idea of this function, thanks for including it in the API.